src/raw/cgroupv2_detect.go

```diff
@@ -0,0 +1,125 @@
+package raw
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"path/filepath"
+	"strings"
+
+	"github.com/newrelic/infra-integrations-sdk/log"
+)
+
+const (
+	// cgroup2 group names patterns for systemd and cgroupfs, where the parameter is the long container ID.
+	// check <https://docs.docker.com/config/containers/runmetrics/#find-the-cgroup-for-a-given-container> for details
+	cgroupV2SystemdTemplate  = "/docker-%s.scope"
+	cgroupV2CgroupfsTemplate = "/%s"
+)
+
+var cgroupV2MountPointNotFoundErr = errors.New("cgroups2 mountpoint was not found")
+
+type cgroupV2Paths struct {
+	// MountPoint is the cgroup2 mount point, Eg: /sys/fs/cgroup
+	MountPoint string
+	// Group is the group path, with the same format that the third parameter in /proc/<pid>/cgroup.
+	// Eg: /system.slice/docker.service
+	Group string
+}
+
+func getCgroupV2Paths(hostRoot string, pid int, driver string, containerID string) (*cgroupV2Paths, error) {
+	mountPoint, err := cgroupV2MountPoint(hostRoot, pid, defaultFileOpenFn)
+	if err != nil {
+		return nil, err
+	}
+	cgroupPath, err := cgroupV2GroupPath(driver, containerID)
+	if err != nil {
+		return nil, err
+	}
+	return &cgroupV2Paths{MountPoint: mountPoint, Group: cgroupPath}, nil
+}
+
+func cgroupV2GroupPath(driver string, containerID string) (string, error) {
+	switch driver {
+	case CgroupSystemd:
+		return fmt.Sprintf(cgroupV2SystemdTemplate, containerID), nil
+	case CgroupGroupfs:
+		return fmt.Sprintf(cgroupV2CgroupfsTemplate, containerID), nil
+	}
+	return "", fmt.Errorf("invalid cgroup2 driver %q", driver)
+}
+
+// cgroupV2MountPoint returns the cgroup mount point which is built joining info from mountsFile (Eg: /proc/mounts)
+// and pid's cgroup file (Eg: /proc/<pid>/cgroup)
+func cgroupV2MountPoint(hostRoot string, pid int, fileOpen fileOpenFn) (string, error) {
```

Comment:
I am not proud of this implementation because it has a lot of code from `cgroupv1_detect` duplicated. I think we should check if this can be avoided, but doing it now (by changing the cgroupv1 detect implementation too) can be more risky than doing it once the cgroup2 support feature is more advance. So, I created the #99 issue so we do not forget it.